lightning/src/ln/channelmanager.rs

```diff
@@ -9538,13 +9547,14 @@ where
 							}
 						});
 					match response {
-						Ok(invoice) => Some(invoice),
-						Err(error) => Some(error),
+						Ok(invoice) => return responder.respond(invoice),
+						Err(error) => return responder.respond(error),
 					}
-				}
+				};
```

Comment:
semi-colon not needed

Reply:
Cleaned up in [77c393a](https://github.com/lightningdevkit/rust-lightning/commit/77c393a6eec06158cbe5fe33608fbe00e2386135)
Thanks!